beacon_node/beacon_chain/src/schema_change/migration_schema_v22.rs

```diff
@@ -0,0 +1,86 @@
+use crate::beacon_chain::BeaconChainTypes;
+use slog::error;
+use slog::{info, Logger};
+use std::sync::Arc;
+use store::chunked_iter::ChunkedVectorIter;
+use store::{
+    chunked_vector::BlockRoots, get_key_for_col, DBColumn, Error, HotColdDB, KeyValueStoreOp,
+};
+use types::Slot;
+
+const LOG_EVERY: usize = 200_000;
+
+pub fn upgrade_to_v22<T: BeaconChainTypes>(
+    db: Arc<HotColdDB<T::EthSpec, T::HotStore, T::ColdStore>>,
+    log: Logger,
+) -> Result<Vec<KeyValueStoreOp>, Error> {
+    info!(log, "Upgrading from v21 to v22");
+
+    let anchor = db.get_anchor_info().ok_or(Error::NoAnchorInfo)?;
+    let split_slot = db.get_split_slot();
+
+    if !db.get_config().allow_tree_states_migration && !anchor.no_historic_states_stored(split_slot)
+    {
+        error!(
+            log,
+            "You are attempting to migrate to tree-states but this is a destructive operation. \
+             Upgrading will require FIXME(sproul) minutes of downtime before Lighthouse starts again. \
+             All current historic states will be deleted. Reconstructing the states in the new \
+             schema will take up to 2 weeks. \
+             \
+             To proceed add the flag --allow-tree-states-migration OR run lighthouse db prune-states"
+        );
+        return Err(Error::DestructiveFreezerUpgrade);
+    }
+
+    let mut ops = vec![];
+
+    rewrite_block_roots::<T>(&db, anchor.oldest_block_slot, split_slot, &mut ops, &log)?;
+
+    let mut genesis_state = db
+        .load_cold_state_by_slot(Slot::new(0))?
+        .ok_or(Error::MissingGenesisState)?;
```

Comment:
This doesn't work. This will try to load the genesis state as a snapshot, but it hasn't been stored as a snapshot yet.

I'll try to fix it shortly.

Reply:
Fixed